sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -12,12 +12,73 @@
 VDM_UNFREEZE = 0
 SYSLOG_IDENTIFIER = "CCmisApi"
 
+VDM_KEY_TO_DB_KEY_PREFIX_MAP = {
+    'Modulator Bias X/I [%]' : 'biasxi',
+    'Modulator Bias X/Q [%]' : 'biasxq',
+    'Modulator Bias X_Phase [%]' : 'biasxp',
+    'Modulator Bias Y/I [%]' : 'biasyi',
+    'Modulator Bias Y/Q [%]' : 'biasyq',
+    'Modulator Bias Y_Phase [%]' : 'biasyp',
+    'CD high granularity, short link [ps/nm]' : 'cdshort',
+    'CD low granularity, long link [ps/nm]' : 'cdlong',
+    'DGD [ps]' : 'dgd',
+    'SOPMD [ps^2]' : 'sopmd',
+    'PDL [dB]' : 'pdl',
+    'OSNR [dB]' : 'osnr',
+    'eSNR [dB]' : 'esnr',
+    'CFO [MHz]' : 'cfo',
+    'Tx Power [dBm]' : 'txcurrpower',
+    'Rx Total Power [dBm]' : 'rxtotpower',
+    'Rx Signal Power [dBm]' : 'rxsigpower'
+}
+
+VDM_SUBTYPE_IDX_MAP= {
+    1: 'highalarm',
+    2: 'lowalarm',
+    3: 'highwarning',
+    4: 'lowwarning',
+    5: 'highalarm_flag',
+    6: 'lowalarm_flag',
+    7: 'highwarning_flag',
+    8: 'lowwarning_flag'
+}
+
+
 helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
 
 class CCmisApi(CmisApi):
     def __init__(self, xcvr_eeprom):
         super(CCmisApi, self).__init__(xcvr_eeprom)
 
+    def _update_dict_if_vdm_key_exists(self, dict_to_be_updated, new_key, vdm_dict_key, vdm_subtype_index, lane=1):
+        '''
+        This function updates the dictionary with the VDM value if the vdm_dict_key exists.
+        @param dict_to_be_updated: the dictionary to be updated.
+        @param new_key: the key to be added in dict_to_be_updated.
+        @param vdm_dict_key: lookup key in the VDM dictionary.
+        @param vdm_subtype_index: the index of the VDM subtype in the VDM page.
+            0 refers to the real time value
+            1 refers to the high alarm threshold
+            2 refers to the low alarm threshold
+            3 refers to the high warning threshold
+            4 refers to the low warning threshold
+            5 refers to the high alarm flag
+            6 refers to the low alarm flag
+            7 refers to the high warning flag
+            8 refers to the low warning flag
+        @param lane: the lane number. Default is 1.
+
+        @return: True if the key exists in the VDM dictionary, False if not.
+        '''
+        try:
+            dict_to_be_updated[new_key] = self.vdm_dict[vdm_dict_key][lane][vdm_subtype_index]
+        except KeyError:
+            dict_to_be_updated[new_key] = 'NA'
```

Comment:
Would reporting NA become an issue when in xcvrd you want to convert real time PM fields into float type?

Reply:
For displaying real time PM fields, this code will not get executed.
https://github.com/sonic-net/sonic-platform-common/blob/796e89a8279681391ab8a125df9ed4e5eb0af126/sonic_platform_base/sonic_xcvr/api/public/c_cmis.py#L848

However, this function seems to be causing an issue for `show int transceiver pm` CLI since it is currently using TRANSCEIVER_DOM_THRESHOLD table along with TRANSCEIVER_PM table to display the output.
I will discuss with @prgeor and update this accordingly so that this CLI doesn't break with this change.

Reply:
@qinchuanares - I have now created the below PR to handle this.
https://github.com/sonic-net/sonic-utilities/pull/3174